src/main/java/rx/Observable.java

```diff
@@ -3666,13 +3675,16 @@ public final Boolean call(T first, T second) {
      *  <dt><b>Scheduler:</b></dt>
      *  <dd>{@code cache} does not operate by default on a particular {@link Scheduler}.</dd>
      * </dl>
+     * <p>
+     * <em>Note:</em> The capacity hint is not an upper bound on cache size. For that, consider
+     * {@link #replay(int)} in combination with {@link ConnectableObservable#autoConnect()} or similar.
      * 
      * @param capacityHint hint for number of items to cache (for optimizing underlying data structure)
      * @return an Observable that, when first subscribed to, caches all of its items and notifications for the
      *         benefit of subsequent subscribers
      * @see <a href="http://reactivex.io/documentation/operators/replay.html">ReactiveX operators documentation: Replay</a>
      */
-    public final Observable<T> cache(int capacityHint) {
+    public final Observable<T> cacheWithCapacityHint(int capacityHint) {
```

Comment:
My inclination would be to leave it plain, since we're deprecating the original name.

It'd be great to see this rename carried into 2.x, or even see this form of the operator dropped altogether.